src/tribler/gui/tribler_window.py

```diff
@@ -545,7 +545,6 @@ def tray_show_message(self, title, message):
     def on_core_connected(self, version):
         if self.core_connected:
             self._logger.warning("Received duplicate Tribler Core connected event")
-            return
```

Comment:
Removing `return` is necessary to successfully re-connect to the new Core when the crash of the previous Core happened right after the GUI was already connected to the Core but before the settings were successfully received.